autoendpoint/src/error.rs

```diff
@@ -152,6 +152,58 @@ impl ApiErrorKind {
         }
     }
 
+    /// Specify the label to use for metrics reporting.
+    pub fn metric_label<'a>(&self) -> &'a str {
```

Comment:
Can simplify this, it is and probably always will be `'static`

```suggestion
    pub fn metric_label(&self) -> &'static str {
```